core/cache/lxdprofilewatcher_test.go

```diff
@@ -142,30 +142,30 @@ func (s *lxdProfileWatcherSuite) TestMachineLXDProfileWatcherSubordinateWithProf
 	defer workertest.CleanKill(c, s.assertStartOneMachineWatcher(c))
 	// Add a new subordinate unit with a profile of a new application.
 	s.newUnitForMachineLXDProfileWatcherSubProfile(c, s.machine0.Id(), unitChange.Name)
-	s.assertChangeValidateMetrics(c, s.wc0.AssertOneChange, 0, 1, 0)
+	s.assertChangeValidateMetrics(c, s.wc0.AssertOneChange, 0, 1, 1)
```

Comment:
All these places, where the miss count is +1 are the result of a subtle bug.

https://github.com/juju/juju/pull/10300/files#diff-ac7fd90596736d93384121b9b5aaaa00L117 would cause a message to be published with an empty model name-space for the first addition of an application. This means it would not reach the LXD profile watcher.

Now that the name-spacing is gone the message does go through and for these tests, result in an extra process-but-do-not-notify step.

Reply:
Is the "extra process-but-do-not-notify step" then a no-op?

Reply:
As far as the watch in these scenarios is concerned, yes.

They get an extra message on the `applicationCharmURLChange`, but no notification is required hence the _miss_ incrementation. Note that these metric names have changed in the _develop_ branch.

Reply:
Fine by me.